thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/yaml/YamlDetectionConfigTranslator.java

```diff
@@ -77,7 +81,7 @@ public YamlDetectionConfigTranslator withExistingDetectionConfig(DetectionConfig
    * Fill in common fields of detection config. Properties of the pipeline is filled by the subclass.
    */
   public DetectionConfigDTO generateDetectionConfig() {
-    validateYAML(yamlConfig);
+    detectionValidator.validateConfig(yamlConfig);
```

Comment:
How about calling detectionValidator in   CompositePipelineConfigTranslator.java's constructor. After this refactoring, each YamlDetectionConfigTranslator implementation has a corresponding DetectionConfigValidator to handle different YAML formats. Because YamlDetectionConfigTranslator is loaded dynamically using reflection, it can call the respective validator. 

For example, if we plan to add EntityLevelPipelineConfigTranslator, it can call EntityLevelDetectionValidator to validate the YAML.

Reply:
Yes. I have already made these changes in the translator cleanup PR which I will raise soon.